rust/agama-lib/src/base_http_client.rs

```diff
@@ -21,15 +21,36 @@ use crate::{auth::AuthToken, error::ServiceError};
 ///   }
 /// ```
 pub struct BaseHTTPClient {
-    client: Client,
+    pub client: reqwest::Client,
```

Comment:
Why is this field public? IMHO, `BaseHTTPClient` should offer the API that fulfills our needs, and the `reqwest::Client` should be just an implementation detail. If we want to replace [reqwest](https://docs.rs/reqwest/latest/reqwest/) with something else (e.g., [ureq](https://github.com/algesten/ureq)) we can do it easily (or if reqwest API changes it might be easy to adapt).

Reply:
It enables me [constructing  a mock client like this](https://github.com/openSUSE/agama/pull/1438/files#diff-e2b8db39f9a74a79051108962095c52672b5b2ff59630d24320b96780d9288b6R98-R101
):
```rs
        let mock_client =  BaseHTTPClient {
            base_url: mock_url,
            ..Default::default()
        };
```

I chose this way because I did not want to come up with a special name for it.

Alternatively we can make it private again, remove the Default trait and make a set of specialized constructors, like `new_unauthenticated_with_url` here?

Reply:
I was playing with an idea to implement some traits here and than do necessary changes in behavior in uderlying structs (which can even cover differences in using authorization token or not)